# TCAP
a transfer-learning based Cox proportional hazards network (TCAP) by utilizing an integrated loss function consisted of two modules: the data reconstruction module to ensure learning a representative hidden layer for the input data, and the proportional hazard module to estimate patients’ risks.

# Requirements
tensorflow, python 3.7, lifelines

# Data preparation
In this study, we utilized cancer datasets from the TCGA portal (https://tcga-data.nci.nih.gov/tcga/). All these datasets were downloaded by using the R package “TCGA-assembler”(v1.0.3, (Wei, et al., 2018)), which contains four types of multi-omics data: mRNA, miRNA, DNA methylation, and copy number variation (CNV) data. Here, “mRNA” was RNA sequencing data generated by UNC Illumina HiSeq_RNASeq V2; Level 3, “miRNA” was miRNA sequencing data obtained by BCGSC Illumina HiSeq miRNASeq, DNA methylation data was generated by USC HumanMethylation450, and CNV data that generated by BROAD-MIT Genome wide SNP_6. 

# Usage
DCAP was a framework with three different method.  Before using you should choose the training data in dataset.py, the test.py is used for choosing the hyper-parameters by k-fold CV and allpred.py is used to construct the prediction model by using all data.

# Example
For easy to use, here we give four example data: brca1.csv used in test.py or allpred.py

# Contact
This method is still on progress, any questions can be sent to chaihuachaihua@qq.com
